sphinx/writers/latex.py

```diff
@@ -1161,7 +1161,9 @@ def visit_paragraph(self, node: Element) -> None:
             # (first one is label node)
             pass
         else:
-            self.body.append('\n')
+            # the \hskip0pt\relax is to allow hyphenation of first word of
+            # a paragraph in narrow contexts such as in a table cell
+            self.body.append('\\hskip0pt\\relax\n')
```

Comment:
It is crucial that this is never inserted before "vertical" material, in particular before a list. Else it will create extra vertical space. I hope that visit_paragraph() will always correspond to starting an actual paragraph in TeX sense of the word. But with substitutions I am not sure.

Adding this rather for example at start of each list item like I first considered (due to relevance to `hlist` of #8779) can not work because the item might itself be a list and adding this will, as I mentioned, then create extra whitespace due to entering horizontal mode with no reason, only to leave it immediately for a "vertical" structure.

Took me some time to change all test files, but if there is a doubt here, better to not merge this and find another way.

Reply:
Sorry, this has a bug of suppressing in output a blank line which is needed by TeX to recognize paragraphs. I am correcting this stupidity in next commit, which will abstract the modification into `\sphinxAtStartPar`.